qiskit_experiments/framework/base_analysis.py

```diff
@@ -167,7 +169,7 @@ def run_analysis(expdata):
             if analysis_results:
                 expdata.add_analysis_results(analysis_results)
             if figures:
-                expdata.add_figures(figures)
+                expdata.add_figures(figures, figure_names=self.options.figure_names)
```

Comment:
In `DbExperimentDataV1.add_figures()` the `figure_names` parameter is documented as either a string or a list. However the method code accepts only None or list for `figure_names, not a string literal. This mandates the user to set figure names only as a list even if it's one name. For example:
``` python
exp1.analysis.set_options(figure_names=["TestFigName"])
```

Please fix `DbExperimentDataV1.add_figures()` to honor the apidoc of the `figure_names` parm

Reply:
`add_figures` technically could accept strings for `figure_names`, but only if also `figures` was a string (which is never the case). I fixed it by replacing the order of some lines